File: Course Handicap Calculator/ViewControllers/CourseHandicapCalculatorVC.m
#import "CourseHandicapCalculatorVC.h"
#import "LabeledTextFieldView.h"
#import "CourseHandicapCalculator.h"


@interface CourseHandicapCalculatorVC ()

@property (strong, nonatomic) IBOutletCollection(UIView) NSArray *transparentViews;

@property (strong, nonatomic) IBOutlet UIView *outputContainer;

@property (strong, nonatomic) IBOutlet LabeledTextFieldView *handicapIndexLabeledTextField;
@property (strong, nonatomic) IBOutlet LabeledTextFieldView *courseSlopeLabeledTextField;
@property (strong, nonatomic) IBOutlet UIButton *calculateButton;

@property (strong, nonatomic) IBOutlet UILabel *yourCourseHandicapLabel;
@property (strong, nonatomic) IBOutlet UILabel *courseHandicapResultLabel;

@end


@implementation CourseHandicapCalculatorVC

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self configureViews];
}

- (void)configureViews {
    [self configureTransparentViews];
    [self hideOutputContainer];
    
    [self prepareLocalization];
}

- (void)configureTransparentViews {
    for (UIView *view in self.transparentViews) {
        view.backgroundColor = [UIColor clearColor];
    }
}

- (void)hideOutputContainer {
    self.outputContainer.hidden = YES;
}

- (void)prepareLocalization {
    self.handicapIndexLabeledTextField.label.text = NSLocalizedString(@"Calculator.Label.HandicapIndex", nil);
    self.courseSlopeLabeledTextField.label.text = NSLocalizedString(@"Calculator.Label.CourseSlope", nil);
    
    self.yourCourseHandicapLabel.text = NSLocalizedString(@"Calculator.Label.YourCouseHandicap", nil);
    
    NSString *caltucateTitle = NSLocalizedString(@"Calculator.Button.Calculate", nil);
    [self.calculateButton setTitle:caltucateTitle forState:UIControlStateNormal];
}

- (IBAction)calculateButtonTapped:(UIButton *)sender {
    [self.view endEditing:YES];

    NSNumber *handicapIndex = @([self.handicapIndexLabeledTextField.textField.text doubleValue]);
    NSNumber *courseSlope = @([self.courseSlopeLabeledTextField.textField.text doubleValue]);
    
    NSInteger courseHandicap = [CourseHandicapCalculator courseHandicapWithHandicapIndex:handicapIndex courseSlope:courseSlope];
    
    self.outputContainer.hidden = NO;
    self.courseHandicapResultLabel.text = @(courseHandicap).stringValue;
}

@end

File: Course Handicap Calculator/ViewControllers/CourseHandicapCalculatorVC.swift
import UIKit


class CourseHandicapCalculatorVC: UIViewController {

    @IBOutlet var transparentViews: [UIView]!
    @IBOutlet var outputContainer: UIView!
    
    @IBOutlet var handicapIndexLabeledTextField: LabeledTextFieldView!
    @IBOutlet var courseSlopeLabeledTextField: LabeledTextFieldView!

    @IBOutlet var calculateButton: UIButton!

    @IBOutlet var yourCourseHandicapLabel: UILabel!
    @IBOutlet var courseHandicapResultLabel: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.configureViews()
    }

    private func configureViews() {
        self.configureTransparentViews()
        self.hideOutputContainer()
        
        self.prepareLocalization()
    }

    private func configureTransparentViews() {
        for view in self.transparentViews {
            view.backgroundColor = UIColor.clear
        }
    }

    private func hideOutputContainer() {
        self.outputContainer.isHidden = true
    }
    
    private func prepareLocalization() {
        self.handicapIndexLabeledTextField.label.text = NSLocalizedString("Calculator.Label.HandicapIndex", comment:"");
        self.courseSlopeLabeledTextField.label.text = NSLocalizedString("Calculator.Label.CourseSlope", comment:"");
    
        self.yourCourseHandicapLabel.text = NSLocalizedString("Calculator.Label.YourCouseHandicap", comment:"");
    
        let caltucateTitle = NSLocalizedString("Calculator.Button.Calculate", comment:"");
        self.calculateButton.setTitle(caltucateTitle, for:.normal)
    }

    @IBAction func calculateButtonTapped(_ sender: UIButton) {
        self.view.endEditing(true)
        
        let handicapIndex = Double(self.handicapIndexLabeledTextField.textField.text!)
        let courseSlope = Double(self.courseSlopeLabeledTextField.textField.text!)
        
        let courseHandicap = CourseHandicapCalculator.courseHandicapWithHandicapIndex(handicapIndex, courseSlope:courseSlope)
        
        self.outputContainer.isHidden = false
        self.courseHandicapResultLabel.text = String(courseHandicap)
    }

}
